### DISCUSSION TITLE
CKEditor text fields as own entity in Field Entries (graphql, headless)

### DISCUSSION THREAD

**Discussion by burningchair** [craftcms/ckeditor#232] [Ideas] [answered]
I would like to use the ckeditor field in the same way as the matrix field. It is possible to fetch all the nested entries with graphql, but not the html between nested entries. e.g. I have a `page` with a ckeditor field `richText` and a `gallery` entryType that can be used in `richText`:
```
query PageQuery($id: [QueryArgument]) {
  entry(type: "page", id: $id) {
    ... on page_Entry {
      id
      uri                  
      richText     
    }
  }
  richTextFieldEntries(ownerId: $id) {    
    __typename
    ... on EntryInterface {
      id
      title
    }
   ... on gallery_Entry {
      images {
        id
        url
        alt
        title
      }
      galleryVariant
    }
}
```
Right now there is no clean solution for rendering the whole richText with own (react) components. **Is it possible to add the text between nested components as own field? e.g.** `richText_text_Entry`? The query for richTextFieldEntries would also include content/text/body field with html:
```
... on richText_text_Entry {
   content
}
```

I know that you could use  `<div data-entry-id="{{entry.id}}"></div>` for the partials but using html-node-parser in my frontend for parsing/replacing these fields is not a good idea.

Right now I am forced to use the matrix field with the ckeditor without nested fields to get all the information as clean json. The authoring experience would be better if i could use only the ckeditor with all the other entry types as nested fields.

**Comment by brandonkelly** [comment, marked as answer]
From your GraphQL schema, you can select `Query for entries in the “[Field Name]” CKEditor field` under “Entries”, and that field’s nested entries will be accessible via `entries` and `entry` queries. You can target them using the `field` param:

```graphql
query {
  entries (field: "body") {
    id
    ...on entryTypeHandle_Entry {
      customFieldHandle
    }
  }
}
```

**Reply by brandonkelly** [reply]
Note there’s a bug where `entries` and `entry` queries will only be available if at least one section’s schema component is selected as well. Fixed that for the next release (craftcms/cms@2dc9c102f2bc89f577363616d651f6a29aac98df).

**Reply by burningchair** [reply]
Thanks for the reply but that is not what I meant. I would need the html around nested elements as (new) own ckeditor html field eg:  lorem-ipsum page with a ckeditor field that includes a image gallery:
![screenshot](https://github.com/craftcms/ckeditor/assets/132882547/b96a4260-e9d5-4eec-942b-dad7e1c5c765)


```
query MyQuery {
  entries (slug: "lorem-ipsum") {    
    id
    ... on page_Entry {
      richText      
  	}        
  }
  richTextFieldEntries (ownerId: 81) {
    __typename
  }
}
```
returns
```
{
  "data": {
    "entries": [
      {
        "id": "81",
        "richText": "<h1>Lorem ipsum dolor sit amet</h1><p>consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.</p><div data-section=\"gallery\" data-id=\"83\"></div><p>At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</p>"
      }
    ],
    "richTextFieldEntries": [
      {
        "__typename": "gallery_Entry"
      }
    ]
  }
}
```

I would need the html content of the ckeditor field before/after the gallery field eg
```
"richTextFieldEntries": [
     {
        "__typename": "ckeditor_html"
      }, 
      {
        "__typename": "gallery_Entry"
      },
      {
        "__typename": "ckeditor_html"
      }
    ]
```

Is this somehow possible right now?
